### ISSUE TITLE
Hyena re-identification case study: revive with a fixed split and metric learning (DL2026 Phase 4, dropped)

### ISSUE THREAD

**Issue by yoavram** [yoavram/DataSciPy#7]
## Background

`sessions/finetuning.ipynb` was a fine-tuning case study on **hyena re-identification** —
`EfficientNetV2S` fine-tuned to classify individual hyenas from the Wild-Me / LILA COCO
export. It was **removed from the `DL2026` branch** rather than reworked: 256 open-set
identities with roughly 12 images each is the wrong task for a first transfer-learning
demonstration, and `sessions/transfer.ipynb` (CUB-200-2011) took the teaching slot.

That was the right call for the course, but the notebook was interesting for a reason
worth keeping: it is a realistic, *hard* re-identification problem whose failure modes are
exactly the ones that make published wildlife re-ID results unreliable. This issue records
what was wrong with it and what would have to be true to bring it back.

Recover the notebook with:

```bash
git show origin/DeepLearning:sessions/finetuning.ipynb > finetuning.ipynb
```

The dataset is a 3.2 GB archive (no longer fetched by `download_data.py`):
`http://us-west-2.opendata.source.coop.s3.amazonaws.com/agentmorris/lila-wildlife/wild-me/hyena.coco.tar.gz`

## Recorded results, as it stood

3,129 images, 256 identities, 2,816 train / 313 validation. Head-only for 5 epochs reached
29.7% validation accuracy; a further 50 epochs of fine-tuning reached 54.0% (loss 2.06).
About 45 minutes of runtime.

**Treat the 54% as unreliable.** See defect 1.

## Five defects, in order of how much they matter

1. **The train/validation split leaks.** `train_test_split` runs over *annotation* indices
   rather than *image ids*. Multiple annotations share an `image_id`, so crops from the
   same photograph can land on both sides of the split, and the model is scored partly on
   photographs it trained on. Split by `image_id` at minimum, and by encounter or date if
   the metadata supports it — same-day sightings are near-duplicates of each other. The
   54% is optimistic by an unknown amount.

   This is worth keeping *visible* in the narrative rather than quietly fixing: it is the
   single most common reason wildlife re-ID papers overstate their results.

2. **The metrics are too weak for the evaluation set.** 313 validation images across 256
   classes is about 1.2 per class, so top-1 accuracy has a standard error near 2.8% and
   many classes have no validation example at all. Report **rank-1 and rank-5**, and a
   **CMC curve**, which is the standard in re-ID and far more informative here.

3. **Viewpoint is unmodelled.** A hyena's left and right flanks carry *different* spot
   patterns, so a single identity label covers two visually distinct classes and a 256-way
   softmax cannot represent that. Check whether the annotations carry a `viewpoint` field
   (Wild-Me COCO exports usually do); if so, report accuracy split by viewpoint and discuss
   why the label space is wrong.

4. **Horizontal flip must never be used here.** It manufactures a fake opposite flank,
   which is a different animal as far as the spot pattern is concerned. Worth an explicit
   cell: the default augmentation pipeline everyone reaches for is *actively wrong* for
   this data.

5. **It was undertrained, not overfit.** Validation loss was still falling monotonically at
   epoch 50, so the reported number is a floor rather than a converged result. Any revival
   should train to convergence before drawing conclusions.

## The interesting follow-up: metric learning

The reason this case study existed was to motivate metric learning, which a 256-way softmax
cannot do:

- **ArcFace** (or another margin-based loss) instead of a softmax classifier.
- **Cosine nearest-neighbour retrieval** against a gallery of embeddings, which is how
  re-identification actually works in the field.
- An **open-set split with held-out individuals** — identities never seen in training — since
  the real task is "have I seen this animal before?", not "which of these 256 is it?".

That is the version worth teaching. It is also a substantially larger piece of work than the
notebook that was cut, and it needs the split fixed first, or it will measure the same
leak with a fancier loss.

## If revived, where does it go?

It cannot simply return as a Day 3 bonus. Day 3 is already full at 6.5 academic hours, and
`sessions/transfer.ipynb` covers probe-versus-fine-tune properly. Options worth considering:

- a standalone bonus notebook, linked from the index but not timetabled;
- material for Day 4, which currently runs about 2 academic hours short of its budget;
- an exercise rather than a session, with the split bug left in as the thing to find.

The last option is tempting: "here is a notebook reporting 54% — find out why that number is
wrong" is a better exercise than most.

**Comment by yoavram** [comment]
Will do a full metric learning experience with two notebooks in #11 and #12 
Closing.
